# Generated by Django 4.2.20 on 2025-04-15 08:35

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='ErrorReport',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('report_type', models.CharField(choices=[('BUG', 'Yazılım Hatası'), ('UI_UX', 'Arayüz/Kullanıcı Deneyimi Sorunu'), ('FUNCTIONALITY', 'İşlevsellik Sorunu'), ('PERFORMANCE', 'Performans Sorunu'), ('OTHER', 'Diğer')], default='OTHER', max_length=20, verbose_name='Hata Türü')),
                ('description', models.TextField(help_text='Lütfen karşılaştığınız hatayı detaylı bir şekilde açıklayın.', verbose_name='Açıklama')),
                ('created_at', models.DateTimeField(auto_now_add=True, verbose_name='Bildirim Zamanı')),
                ('resolved', models.BooleanField(default=False, verbose_name='Çözüldü mü?')),
                ('page_url', models.URLField(blank=True, max_length=1024, null=True, verbose_name="Hatanın Olduğu Sayfa URL'si")),
                ('user', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='error_reports', to=settings.AUTH_USER_MODEL, verbose_name='Bildiren Kullanıcı')),
            ],
            options={
                'verbose_name': 'Hata Bildirimi',
                'verbose_name_plural': 'Hata Bildirimleri',
                'ordering': ['-created_at'],
            },
        ),
    ]
